### PR TITLE
New integration - CounterTack

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
In Progress

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/integration-CounterTack.yml [hunk 1]
```diff
@@ -0,0 +1,1716 @@
...
+description: CounterTack empowers endpoint security teams to assure endpoint protection
+  for Identifying Cyber Threats. Integrating a predictive endpoint protection platform
+configuration:
+  - display: Server URL
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please add example, like `(e.g. https://democloud.countertack.com)`

File: Integrations/integration-CounterTack.yml [hunk 2]
```diff
@@ -0,0 +1,1716 @@
...
+    name: isFetch
+    defaultvalue: ""
+    type: 8
+    required: false
+  - display: Incident type
+    name: incidentType
+    defaultvalue: ""
+    type: 13
+    required: false
+  - display: Use system proxy settings
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: this parameter should be placed above the insecure one

File: Integrations/integration-CounterTack.yml [hunk 3]
```diff
@@ -0,0 +1,1716 @@
...
+    import json
+    import requests
+    import os
+    import base64
+
+    # Disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    # PREREQUISITES
+    def load_server_url():
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: there isn't a real need for this function, please use our standard way to just get rid of the trailing slash

File: Integrations/integration-CounterTack.yml [hunk 4]
```diff
@@ -0,0 +1,1716 @@
...
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    ''' GLOBALS/PARAMS '''
+
+    USERNAME = demisto.params().get('credentials').get('identifier')
+    PASSWORD = demisto.params().get('credentials').get('password')
+    SERVER_URL = load_server_url()
+    # Should we use SSL
+    USE_SSL = not demisto.params()['insecure']
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please use `.get()` and add default `False` value

File: Integrations/integration-CounterTack.yml [hunk 5]
```diff
@@ -0,0 +1,1716 @@
...
+        try:
+            response = requests.request(
+                method,
+                url,
+                auth=(USERNAME,PASSWORD),
+                headers=headers,
+                verify=USE_SSL,
+                data=body
+        )
+        except requests.exceptions.SSLError as e:
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: why is this needed ?

File: Integrations/integration-CounterTack.yml [hunk 6]
```diff
@@ -0,0 +1,1716 @@
...
+            LOG(e.message)
+            raise ValueError('An SSL error occurred when trying to connect to the server. Consider configuring unsecure connection in the integration settings.')
+
+        #handle request failure
+        if response.status_code not in range(200,205):
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: are you sure the range 200-205 is correct?

File: Integrations/integration-CounterTack.yml [hunk 7]
```diff
@@ -0,0 +1,1716 @@
...
+            message = parse_error_response(response)
+            raise ValueError('Request failed with status code {}\n{}'.format(response.status_code, message))
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: use `return_error` instead of raise
rephrase the error message to be something like `Error in API call to CounterTack...`

File: Integrations/integration-CounterTack.yml [hunk 8]
```diff
@@ -0,0 +1,1716 @@
...
+        return response
+
+
+    def parse_error_response(response):
+        try:
+            res = response.json()
+            msg = res.get('message')
+            if res.get('details') is not None and res.get('details')[0].get('message') is not None:
+                msg = msg + "\n"+json.dumps(res.get('details')[0])
+        except Exception as e:
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: you are not using `e`

File: Integrations/integration-CounterTack.yml [hunk 9]
```diff
@@ -0,0 +1,1716 @@
...
+            return response.text
+        return msg
+
+    def return_error_entry(message):
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: this is not needed, please use `return_error` from common server

File: Integrations/integration-CounterTack.yml [hunk 10]
```diff
@@ -0,0 +1,1716 @@
...
+            'Type': entryTypes['error'],
+            'Contents': message,
+            'ContentsFormat': formats['text']
+        }
+        demisto.results(error_entry)
+
+    def behavior_to_incident(behavior):
+            incident = {}
+            incident['name'] = 'CounterTack Behavior Triggered: ' + behavior['name']
+            incident['occurred'] = behavior['last_found'] + 'Z'
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: why are you adding `Z` ?

File: Integrations/integration-CounterTack.yml [hunk 11]
```diff
@@ -0,0 +1,1716 @@
...
+    def return_error_entry(message):
+        error_entry = {
+            'Type': entryTypes['error'],
+            'Contents': message,
+            'ContentsFormat': formats['text']
+        }
+        demisto.results(error_entry)
+
+    def behavior_to_incident(behavior):
+            incident = {}
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: just build the object at once, not key by key

File: Integrations/integration-CounterTack.yml [hunk 12]
```diff
@@ -0,0 +1,1716 @@
...
+        url = '{}/endpoints'.format(BASE_PATH)
+        response = http_request('GET',url,headers=GET_HEADERS)
+        return response.json()
+
+    def get_endpoints():
+
+        """
+        Returns the information on existing endpoints
+        """
+        headers = ['OS','Name','Threat','Status','ID','IP']
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please place this line above the entry object initialization

File: Integrations/integration-CounterTack.yml [hunk 13]
```diff
@@ -0,0 +1,1716 @@
...
+            data.append ({
+                'ID': endpoint.get('id'),
+                'Name': endpoint.get('name'),
+                'OS': endpoint.get('product_name'),
+                'IP': endpoint.get('ips'),
+                'Status': endpoint.get('status'),
+                'Threat': endpoint.get('threat')
+            })
+
+        context['CounterTack.Endpoint(val.ID && val.ID === obj.ID)'] = endpoints
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: `endpoints` have no key `ID`, but `id`
need to process the data for the context like for the human readable

File: Integrations/integration-CounterTack.yml [hunk 14]
```diff
@@ -0,0 +1,1716 @@
...
+    USE_SSL = not demisto.params()['insecure']
+    # Service base URL
+    BASE_PATH = '{}/api/v2'.format(SERVER_URL)
+    # Headers to be sent in requests
+    GET_HEADERS = {
+        'Content-Type': 'application/json'
+    }
+
+
+    def http_request(method, url, headers={}, body=None, body_params=None):
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: make the `headers` default to be `GET_HEADERS` instead of empty dict, and remove it from all requests functions

File: Integrations/integration-CounterTack.yml [hunk 15]
```diff
@@ -0,0 +1,1716 @@
...
+        """
+        returns the http request
+
+        """
+        #basic authentication
+        try:
+            response = requests.request(
+                method,
+                url,
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: do the formatting of adding the `BASH_PATH` here and not in every request function

File: Integrations/integration-CounterTack.yml [hunk 16]
```diff
@@ -0,0 +1,1716 @@
...
+            })
+
+        context['CounterTack.Endpoint(val.ID && val.ID === obj.ID)'] = endpoints
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': endpoints,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('CounterTack Endpoints information:', data, headers),
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please improve title, to something like `CounterTack Endpoints`. No need in `infromation` and `:`

File: Integrations/integration-CounterTack.yml [hunk 17]
```diff
@@ -0,0 +1,1716 @@
...
+        headers = ['OS','Domain','IP','Threat','Max_impact','Tenant_ID','Is_Quarantine','Profile','Tags','Cluster_Hosts','Status',]
+        endpoint_id = demisto.args().get('endpoint_id')
+        response = get_endpoint_request(endpoint_id)
+
+        content = {
+            'OS': response.get('product_name'),
+            'Domain': response.get('domain'),
+            'IP': response.get('ip'),
+            'Threat': response.get('threat'),
+            'Max_impact': response.get('max_impact'),
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: Should be `MaxImpact`

File: Integrations/integration-CounterTack.yml [hunk 18]
```diff
@@ -0,0 +1,1716 @@
...
+            'Tenant_ID': response.get('tenant'),
+            'Is_Quarantine': response.get('is_quarantined'),
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: should be `IsQuarantined`

File: Integrations/integration-CounterTack.yml [hunk 19]
```diff
@@ -0,0 +1,1716 @@
...
+        endpoint_id = demisto.args().get('endpoint_id')
+        response = get_endpoint_request(endpoint_id)
+
+        content = {
+            'OS': response.get('product_name'),
+            'Domain': response.get('domain'),
+            'IP': response.get('ip'),
+            'Threat': response.get('threat'),
+            'Max_impact': response.get('max_impact'),
+            'Tenant_ID': response.get('tenant'),
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: should be `TenantID`

File: Integrations/integration-CounterTack.yml [hunk 20]
```diff
@@ -0,0 +1,1716 @@
...
+            'Tags': response.get('tags')
+        }
+
+        context['CounterTack.Endpoint(val.Tenant_ID && val.Tenant_ID === obj.Tenant_ID)'] = response
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': response,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('CounterTack Endpoint information:', content,headers,removeNull=True),
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please add space between comma and `removeNull`

File: Integrations/integration-CounterTack.yml [hunk 21]
```diff
@@ -0,0 +1,1716 @@
...
+
+
+    def http_request(method, url, headers={}, body=None, body_params=None):
+        """
+        returns the http request
+
+        """
+        #basic authentication
+        try:
+            response = requests.request(
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: make this in one line, no need to break

File: Integrations/integration-CounterTack.yml (hunk 13 above)
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: where is `context` being initialized ?

File: Integrations/integration-CounterTack.yml [hunk 22]
```diff
@@ -0,0 +1,1716 @@
...
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown("CounterTack tags for the specified endpoint:", response),
+            'EntryContext': context
+        }
+        demisto.results(entry)
+
+
+
+    def add_tags_request(endpoint_id,body):
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: add space after comma

File: Integrations/integration-CounterTack.yml [hunk 23]
```diff
@@ -0,0 +1,1716 @@
...
+        })
+
+        context['CounterTack.Endpoint(val.ID && val.ID === obj.ID)'] = tags_context
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': response,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown("CounterTack tags for the specified endpoint:", response),
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: `"` -> `'`

File: Integrations/integration-CounterTack.yml [hunk 24]
```diff
@@ -0,0 +1,1716 @@
...
+
+        tags_context = []
+        endpoint_id = demisto.args().get('endpoint_id')
+        response = endpoint_tags_request(endpoint_id)
+
+        response = {
+            'tags': response
+        }
+
+        tags_context.append({
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: why is `tags_context` a list?

File: Integrations/integration-CounterTack.yml [hunk 25]
```diff
@@ -0,0 +1,1716 @@
...
+
+        endpoint_id = demisto.args().get('endpoint_id')
+        body = argToList(demisto.args().get('tags'))
+        if not isinstance(body, list):
+            body = [body]
+        response = add_tags_request(endpoint_id,body)
+        endpoint_tags = endpoint_tags_request(endpoint_id)
+
+        response = {
+            'tags':endpoint_tags,
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: add space after `:`

File: Integrations/integration-CounterTack.yml [hunk 26]
```diff
@@ -0,0 +1,1716 @@
...
+
+        demisto parameter: (string) endpoint_id
+            The unique ID of the endpoint
+        demisto parameter: (array) body
+            The tags to add to the endpoint
+        """
+
+        endpoint_id = demisto.args().get('endpoint_id')
+        body = argToList(demisto.args().get('tags'))
+        if not isinstance(body, list):
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: this is unnecessary as `argToList` returns a list

File: Integrations/integration-CounterTack.yml [hunk 27]
```diff
@@ -0,0 +1,1716 @@
...
+            body = [body]
+        response = add_tags_request(endpoint_id,body)
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: after space after `,`

File: Integrations/integration-CounterTack.yml [hunk 28]
```diff
@@ -0,0 +1,1716 @@
...
+
+    USERNAME = demisto.params().get('credentials').get('identifier')
+    PASSWORD = demisto.params().get('credentials').get('password')
+    SERVER_URL = load_server_url()
+    # Should we use SSL
+    USE_SSL = not demisto.params()['insecure']
+    # Service base URL
+    BASE_PATH = '{}/api/v2'.format(SERVER_URL)
+    # Headers to be sent in requests
+    GET_HEADERS = {
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: i see you use these headers for all HTTP methods, so these are not `GET` headers, but `DEFAULT_HEADERS`

File: Integrations/integration-CounterTack.yml [hunk 29]
```diff
@@ -0,0 +1,1716 @@
...
+            'HumanReadable': tableToMarkdown("Endpoint tags were added successfully", response),
+            'EntryContext': context
+        }
+        demisto.results(entry)
+
+    def delete_tags_request(endpoint_id,body):
+
+        """
+        This request deletes specific tags from specified endpoint.
+        The request gets the endpoint ID and the tags to be deleted from it
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: the arguments docstring is not well formatted

File: Integrations/integration-CounterTack.yml [hunk 30]
```diff
@@ -0,0 +1,1716 @@
...
+
+        demisto parameter: (string) endpoint_id
+            The unique ID of the endpoint
+        demisto parameter: (string) body
+            The type of the command: extract file and the file path
+        """
+
+        endpoint_id = demisto.args().get('endpoint_id')
+
+        args = demisto.args()
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: why are you doing all of a sudden? need to be consistent in all functions, so please remove this

File: Integrations/integration-CounterTack.yml [hunk 31]
```diff
@@ -0,0 +1,1716 @@
...
+        body = {
+            'type': args.get('type'),
+            'file_paths': argToList(args.get('paths'))
+        }
+        response = file_extract_request(endpoint_id,body)
+
+        data = {
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: Please format keys according to convention

File: Integrations/integration-CounterTack.yml [hunk 32]
```diff
@@ -0,0 +1,1716 @@
...
+            'command name': response.get('command_name'),
+            'command arguments': response.get('command_arg'),
+            'status': response.get('status'),
+        }
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': response,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('The file has been extracted successfully:', data),
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: missing headers

File: Integrations/integration-CounterTack.yml [hunk 33]
```diff
@@ -0,0 +1,1716 @@
...
+            'EntryContext': {
+                "CounterTack.EndpointCommand(obj.ID===val.ID)": response
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: don't set the key here, but before the entry

File: Integrations/integration-CounterTack.yml [hunk 34]
```diff
@@ -0,0 +1,1716 @@
...
+        response = http_request(
+            'GET',
+            url,
+            headers = GET_HEADERS
+        )
+        return response.json()
+
+    def get_all_files():
+
+        headers = ['Status','ID','path','endpoint_id','extraction_time','user']
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: add spaces after comma

File: Integrations/integration-CounterTack.yml [hunk 35]
```diff
@@ -0,0 +1,1716 @@
...
+            'status': response.get('status'),
+        }
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': response,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Endpoint information:', data, headers),
+            'EntryContext': {
+                "CounterTack.Files(obj.ID===val.ID)": response
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: add `val.ID && `

File: Integrations/integration-CounterTack.yml [hunk 36]
```diff
@@ -0,0 +1,1716 @@
...
+            'Contents': response,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Endpoint information:', data),
+            'EntryContext': {
+                "CounterTack.Files(obj.ID===val.ID)": response
+            }
+        }
+        demisto.results(entry)
+    """
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: remove this

File: Integrations/integration-CounterTack.yml [hunk 37]
```diff
@@ -0,0 +1,1716 @@
...
+            data.append ({
+                'ID': behavior.get('id'),
+                'name': behavior.get('name'),
+                'Type': behavior.get('type'),
+                'Impact_Level': behavior.get('impact_level'),
+                'Last_reported': behavior.get('last_reported'),
+                'Endpoint_ID': behavior.get('endpoint_id')
+        })
+
+        context['CounterTack.Behavior(val.ID && val.ID == obj.ID)'] = behaviors
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: use `===`

File: Integrations/integration-CounterTack.yml [hunk 38]
```diff
@@ -0,0 +1,1716 @@
...
+    """
+
+    FETCH INCIDENTS
+
+    """
+    def fetch_incidents():
+        now = datetime.utcnow()
+        last_run = demisto.getLastRun().get('time')
+        if last_run is None:
+            now = now - timedelta(days=3)
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: the time delta to fetch from should be configurable and not 3 days by default, please add parameter for that

File: Integrations/integration-CounterTack.yml [hunk 39]
```diff
@@ -0,0 +1,1716 @@
...
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('CounterTack Endpoints Behaviors:', data, headers),
+            'EntryContext': context
+        }
+        demisto.results(entry)
+
+    def get_behavior_request(behavior_id):
+        """
+        Request for getting  specified behvior
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: remove extra white space after `getting`

File: Integrations/integration-CounterTack.yml [hunk 40]
```diff
@@ -0,0 +1,1716 @@
...
+    def fetch_incidents():
+        now = datetime.utcnow()
+        last_run = demisto.getLastRun().get('time')
+        if last_run is None:
+            now = now - timedelta(days=3)
+            last_run = now
+        else:
+            last_run = datetime.strptime(last_run, '%Y-%m-%dT%H:%M:%S') #Converts string to datetime object
+        current_run = last_run
+        behaviors = get_behavior_request()
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: `get_behavior_request` expects argument `behavior_id`, and you when don't pass anything it will just concat `None` to the URL - is that ok?
Is there no API query to get behaviors by timestamp ?

File: Integrations/integration-CounterTack.yml [hunk 41]
```diff
@@ -0,0 +1,1716 @@
...
+            if last_found > last_run:
+                incident = behavior_to_incident(behavior)
+                incidents.append(incident)
+            if last_found > current_run:
+                current_run = last_found
+
+        demisto.incidents(incidents)
+        demisto.setLastRun({ 'time': current_run.isoformat().split('.')[0] })
+
+
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: too many empty lines

File: Integrations/integration-CounterTack.yml [hunk 42]
```diff
@@ -0,0 +1,1716 @@
...
+        for behavior in behaviors:
+            last_found = datetime.strptime(behavior['last_found'], '%Y-%m-%dT%H:%M:%S')
+            if last_found > last_run:
+                incident = behavior_to_incident(behavior)
+                incidents.append(incident)
+            if last_found > current_run:
+                current_run = last_found
+
+        demisto.incidents(incidents)
+        demisto.setLastRun({ 'time': current_run.isoformat().split('.')[0] })
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: not sure you're handling the timestamp right, can you explain what you're trying to do ?
lynt properly

File: Integrations/integration-CounterTack.yml [hunk 43]
```diff
@@ -0,0 +1,1716 @@
...
+        elif command == 'countertack-get-behaviors':
+            get_behaviors()
+        elif command == 'countertack-get-behavior':
+            get_behavior()
+        elif command == 'countertack-add-behavior-tags':
+            add_behavior_tags()
+        elif command == 'countertack-delete-behavior-tags':
+            delete_behavior_tags()
+    except Exception as e:
+        raise
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: remove

File: Integrations/integration-CounterTack.yml [hunk 44]
```diff
@@ -0,0 +1,1716 @@
...
+          description: IP address used to connect to analysis cluster.
+          type: string
+        - contextPath: CounterTack.Endpoint.cluster_hosts
+          description: List of hosts the endpoint tries to connect through in order.
+          type: string
+        - contextPath: CounterTack.Endpoint.vendor
+          description: OS vendor.
+          type: string
+        - contextPath: CounterTack.Endpoint.sensor_mode
+          description: ' Specifies the sensor mode of the driver.'
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: remove unnecessary spaces from all descriptions

File: Integrations/integration-CounterTack.yml [hunk 45]
```diff
@@ -0,0 +1,1716 @@
...
+          type: string
+        - contextPath: CounterTack.Endpoint.tags
+          description: List of user assigned tags.
+          type: string
+      description: Get information on specific endpoint
+    - name: countertack-get-behavior
+      arguments:
+        - name: behavior_id
+          required: true
+          description: The id of the behavior
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: improved description like you did in previous commands

File: Integrations/integration-CounterTack.yml [hunk 46]
```diff
@@ -0,0 +1,1716 @@
...
+          type: string
+        - contextPath: CounterTack.Behavior.last_reported
+          description: When the behaviour  last seen
+          type: date
+      description: Get information on specific behavior
+    - name: countertack-get-endpoint-tags
+      arguments:
+        - name: endpoint_id
+          required: true
+          description: The id of an endpoint
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: improve description

File: Integrations/integration-CounterTack.yml [hunk 47]
```diff
@@ -0,0 +1,1716 @@
...
+        - contextPath: CounterTack.Endpoint.tags
+          description: The tags that were added to the endpoint
+          type: string
+      description: Add tags for specified endpoint
+    - name: countertack-delete-tags
+      arguments:
+        - name: tags
+          required: true
+          isArray: true
+          description: The tags to be deleted from specified endpoint
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please add that it supports CSV, and add example

File: Integrations/integration-CounterTack.yml [hunk 48]
```diff
@@ -0,0 +1,1716 @@
...
+          type: string
+        - contextPath: CounterTack.Behavior.tags
+          description: The tags of the behavior
+      description: 'Delete specific tags from specified behavior '
+    - name: countertack-endpoint-quarantine
+      arguments:
+        - name: endpoint_id
+          required: true
+          description: The Id of the endpoint
+        - name: type
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: we agreed to remove this argument..

File: Integrations/integration-CounterTack.yml [hunk 49]
```diff
@@ -0,0 +1,1716 @@
...
+            get_endpoint()
+        elif command == 'countertack-get-endpoint-tags':
+            get_endpoint_tags()
+        elif command == 'countertack-add-tags':
+            add_tags()
+        elif command == 'countertack-delete-tags':
+            delete_tags()
+        elif command == 'countertack-endpoint-quarantine':
+            endpoint_quarantine()
+        elif command == 'countertack-lift-quarantine':
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: there is no such command, did you mean `countertack-disable-quarantine` ?

File: Integrations/integration-CounterTack.yml [hunk 50]
```diff
@@ -0,0 +1,1716 @@
...
+        - contextPath: CounterTack.Files.path
+          description: Full file system path of the extracted file, including the filename,
+            as seen on the endpoint.
+          type: string
+        - contextPath: CounterTack.Files.sha256
+          description: The SHA256 digest of the file contents.
+          type: string
+        - contextPath: CounterTack.Files.tenant
+          description: Tenant ID for endpoint.
+
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: no need in `releaseNotes` for new files

File: Integrations/integration-CounterTack.yml (hunk 8 above)
Comment (yardensade) [reply Integrations/integration-CounterTack.yml]: still not handled

File: Integrations/integration-CounterTack.yml [hunk 51]
```diff
@@ -0,0 +1,1693 @@
...
+        demisto parameter: (string) endpoint_id
+            The unique ID of the endpoint
+
+        returns:
+            The information about the specified endpoint
+        """
+        headers = ['OS', 'Domain', 'IP', 'Threat', 'MaxImpact', 'TenantID', 'IsQuarantined', 'Profile', 'Tags', 'Cluster_Hosts', 'Status']
+        endpoint_id = demisto.args().get('endpoint_id')
+        response = get_endpoint_request(endpoint_id)
```
Comment (yardensade) [inline Integrations/integration-CounterTack.yml]: should check if resposne or if response is instance of dict or else the .get() will throw exception

File: Integrations/integration-CounterTack.yml [hunk 52]
```diff
@@ -0,0 +1,1695 @@
...
+                'Memory': endpoint.get('memory'),
+                'Processors': endpoint.get('num_cpus')
+            }
+
+        context = {
+            'CounterTack.Endpoint(val.id && val.id === obj.id)': endpoints
+        }
+
+        context = {
+            'Endpoint(val.id && val.id === obj.id)': endpoint_standards
```
Comment (yardensade) [inline Integrations/integration-CounterTack.yml]: You have assigned twice context, so the second one will run over the first one.
Make sure to append like you did before that (context[]), atleast the second one

File: Integrations/integration-CounterTack.yml [hunk 53]
```diff
@@ -0,0 +1,1695 @@
...
+                'MACAddress': endpoint.get('mac'),
+                'OS': endpoint.get('product_name'),
+                'OSVersion': endpoint.get('driver_version'),
+                'Model': endpoint.get('current_profile'),
+                'Memory': endpoint.get('memory'),
+                'Processors': endpoint.get('num_cpus')
+            }
+
+        context = {
+            'CounterTack.Endpoint(val.id && val.id === obj.id)': endpoints
```
Comment (yardensade) [inline Integrations/integration-CounterTack.yml]: endpoint = endpoint_standards (since you already formatted it)

File: Integrations/integration-CounterTack.yml [hunk 54]
```diff
@@ -0,0 +1,1695 @@
...
+        demisto parameter: (string) endpoint_id
+            The unique ID of the endpoint
+        demisto parameter: (string) type
+            The type of the command: quarantine
+        """
+
+        suffix_url = 'endpoints/' + endpoint_id + '/commands'
+
+        response = http_request('POST', suffix_url, body=json.dumps(body))
+        return response.json()
```
Comment (yardensade) [inline Integrations/integration-CounterTack.yml]: validate response is dict

File: Integrations/integration-CounterTack.yml [hunk 55]
```diff
@@ -0,0 +1,1695 @@
...
+    def file_extract_request(endpoint_id, body):
+
+        """
+        Request for extracting file from specified endpoint
+        """
+
+        suffix_url = 'endpoints/' + endpoint_id + '/commands'
+
+        response = http_request('POST', suffix_url, body=json.dumps(body))
+        return response.json()
```
Comment (yardensade) [inline Integrations/integration-CounterTack.yml]: validate is dict

File: Integrations/integration-CounterTack.yml [hunk 56]
```diff
@@ -0,0 +1,1695 @@
...
+        demisto.results(entry)
+
+
+    def file_information_request(file_id):
+        """
+        request specific file information
+        """
+        suffix_url = 'endpoints/files/' + file_id
+        response = http_request('GET', suffix_url)
+        return response.json()
```
Comment (yardensade) [inline Integrations/integration-CounterTack.yml]: validate dict

File: Integrations/integration-CounterTack.yml [hunk 57]
```diff
@@ -0,0 +1,1695 @@
...
+
+
+    def get_behavior_request(behavior_id):
+        """
+        Request for getting specified behvior
+        """
+        suffix_url = 'behaviors/' + behavior_id
+
+        response = http_request('GET', suffix_url)
+        return response.json()
```
Comment (yardensade) [inline Integrations/integration-CounterTack.yml]: validate is dict

**Comment by Itay4** [comment]
https://github.com/demisto/etc/issues/11952
